ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+            print("Remapped Elapsed time:", t1_stop-t1_start) 
+            print('Data has been regridded for mask2d')
+            rgmask2d = np.where(rgmask2d>0.0,  1.0, 0.0)
+
+        if model == 'MOM':
+            print('beginning the regridding for mask3d')                                                                                                                                                                                                 
+            if regridtype == 'manual':
+               rgmask3d=esmfmanualregrid(mask3d,S1,row1,col1,frac_b1,n_s1,'bilinear',dimyh,dimxh,len(latd),len(lond))
+            print('Data has been regridded for mask3d') 
+            rgmask3d = np.where(rgmask3d>0.0,  1.0, 0.0)       
+
+        #Create Mask NETCDF File#########      
+        testfile='masks_'+dstgrds[jj]+'.nc'
+        os.system('rm -vf '+testfile)
+        f = nc.Dataset(testfile,'w', format='NETCDF4') 
+
+        if model == 'MOM':
+            f.createDimension('z_l', nlevs) 
+        f.createDimension('lon', len(lond))
+        f.createDimension('lat', len(latd))
+        f.createDimension('time', None)
+
+        if model == 'MOM':
+            depth = f.createVariable('z_l', 'f4', 'z_l')
+            depth.long_name='Depth at cell center'
+            depth.units = 'meters'
+
+        longitude = f.createVariable('lon', 'f4', 'lon')
+        longitude.long_name='longitude'
+        longitude.units = 'degrees_east'
+
+        latitude = f.createVariable('lat', 'f4', 'lat')  
+        latitude.long_name = 'latitude'
+        latitude.units = 'degrees_north'
+
+        rgmask2dn = f.createVariable('rgmask2d1', 'f4', ('time', 'lat', 'lon'))
+        if model == 'MOM':
+            rgmask3dn = f.createVariable('rgmask3d', 'f4', ('time', 'z_l','lat', 'lon')) 
+        time = f.createVariable('Time', 'i4', 'time')
+
+        longitude[:] = lond    
+        if model == 'MOM':                                                                                                                                                                                                                                                   
+            depth[:]=z_l[:]
+        latitude[:] = latd
+
+        rgmask2dn[0,:,:] = rgmask2d
+        if model == 'MOM':
+            rgmask3dn[0,:,:,:] = rgmask3d
+
+        f.close()
+
+        #################################################
+
+        for fl in range(0,dimfiles):
+            t1_start = process_time()
+            infile0=infiles1[fl]
+            outfile0=outfiles1[fl]
+            ocnf=nc.Dataset(infile0)
+            print('Input file is '+infile0)
+            print('Output file is '+outfile0)
+        #Extract time
+            intime=ocnf['time']
+                
+        #Create output regridded netcdf file 
+            FILENAME_REGRID = outfile0
+            os.system('rm -f '+FILENAME_REGRID)
+            outcdf = nc.Dataset(FILENAME_REGRID,'w', format='NETCDF3_CLASSIC')
```

Comment:
why are we writing out `NETCDF3` version filed?

Reply:
I believe NETCDF3 was the format that was used in the ocean and ice post NCL programs. However, there is no particular reason we need to keep it this way. I'll change this to NETCDF4, or this can be a user-configurable option in a yaml file.

Reply:
> I believe NETCDF3 was the format that was used in the ocean and ice post NCL programs. However, there is no particular reason we need to keep it this way. I'll change this to NETCDF4, or this can be a user-configurable option in a yaml file.

it is netcdf4

Reply:
Got it, thanks Jiande.

Reply:
netCDF3 is an artifact of NGGPS.